app/components/client-only/client-only.tsx

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from "react";
+import type { ClientOnlyProps } from "./types";
+
+let hydrating = true;
+
+function ClientOnly({ children, fallback = null }: ClientOnlyProps) {
+  const [hydrated, setHydrated] = useState(() => !hydrating);
+
+  useEffect(function hydrate() {
+    hydrating = false;
+    setHydrated(true);
+  }, []);
+
+  return hydrated ? <>{children()}</> : <>{fallback}</>;
+}
```

Comment:
Remix by default will try to do an initial server rendering of all the components and deliver them to the browser, this might be a problem for some components that would depend on any browser API, which would break on the server. 

This component is a HOC that will wrap these components and skip their server rendering, allowing the client to do the whole job. I haven't used it yet, but it might be useful for later